libsweep/src/sweep.cc

```diff
@@ -46,149 +41,102 @@ void sweep_error_destruct(sweep_error_s error) {
   delete error;
 }
 
-sweep_device_s sweep_device_construct_simple(sweep_error_s* error) {
-  SWEEP_ASSERT(error);
-
-  return sweep_device_construct("/dev/ttyUSB0", 115200, error);
-}
-
-sweep_device_s sweep_device_construct(const char* port, int32_t bitrate, sweep_error_s* error) {
+// internal implementations of functions - can throw exceptions
+namespace {
+
+struct Error : ::sweep::ErrorBase {
+  using ::sweep::ErrorBase::ErrorBase;
+};
```

Comment:
That would also be a possibility. 
In general I wanted to distinguish between exceptions thrown in the library in general and exceptions that are thrown on this particular level (e.g. as opposed to the IO or protocol level). But currently there is admittedly nothing that makes use of that distinction. We could even use the same exception type everywhere in the code.